drivers/gnss/gnss_u_blox_m10.c

```diff
@@ -0,0 +1,888 @@
+		frame_data->config_blocks[index].flags = enable |
+			UBX_CFG_GNSS_FLAG_SGN_CNF_SBAS_L1C_A;
+		break;
+	default:
+		return -EINVAL;
+	};
+
+	return 0;
+}
+
+static int ubx_m10_set_enabled_systems(const struct device *dev, gnss_systems_t systems)
+{
+	int ret;
+	k_spinlock_key_t key;
+	struct ubx_m10_data *data = dev->data;
+
+	key = k_spin_lock(&data->lock);
+
+	struct ubx_cfg_gnss_data *frame_data;
+	struct modem_ubx_script script;
+
+	/* Get number of tracking channels for each supported gnss system by sending CFG-GNSS. */
+	ret = ubx_m10_modem_ubx_script_init(dev, &script, data->ubx_frame_buf,
+		sizeof(data->ubx_frame_buf), MODEM_UBX_RETRY_DEFAULT, NULL, UBX_CLASS_CFG,
+		UBX_CFG_GNSS, UBX_FRM_GET_PAYLOAD_SZ);
+	if (ret < 0) {
+		goto out_2;
+	}
+
+	script.ubx_frame_size = ret;
+	ret = ubx_m10_modem_ubx_run_script(dev, &script);
+	if (ret < 0) {
+		goto out_2;
+	}
+
+	struct ubx_frame_t *frame = (struct ubx_frame_t *) script.ubx_frame;
+	uint16_t res_trk_ch_sum = 0, max_trk_ch_sum = 0;
+
+	/* Calculate sum of reserved and maximum tracking channels for each supported gnss system,
+	 * and assert that the sum is not greater than the number of tracking channels in use.
+	 */
+	frame_data = (struct ubx_cfg_gnss_data *) frame->payload_and_checksum;
+	for (int i = 0; i < frame_data->num_config_blocks; ++i) {
+		ret = ubx_m10_ubx_gnss_id_to_gnss_system(dev, frame_data->config_blocks[i].gnss_id);
+		if (ret < 0) {
+			ret = -EINVAL;
+			goto out_2;
+		}
+
+		if (ret & systems) {
+			res_trk_ch_sum += frame_data->config_blocks[i].num_res_trk_ch;
+			max_trk_ch_sum += frame_data->config_blocks[i].max_num_trk_ch;
+		}
+
+		if (res_trk_ch_sum > frame_data->num_trk_ch_use ||
+		    max_trk_ch_sum > frame_data->num_trk_ch_use) {
+			ret = -EINVAL;
+			goto out_2;
+		}
+	}
+
+	/* Prepare frame_data (payload) for sending CFG-GNSS for enabling the gnss systems. */
+	frame_data = malloc(sizeof(*frame_data) +
+		sizeof(struct ubx_cfg_gnss_data_config_block) * UBX_M10_GNSS_SUPP_SYS_CNT);
+	frame_data->num_config_blocks = UBX_M10_GNSS_SUPP_SYS_CNT;
+
+	(void) ubx_cfg_gnss_data_default(frame_data);
+
+	uint8_t filled_blocks = 0;
+	gnss_systems_t supported_systems;
+	(void) ubx_m10_get_supported_systems(dev, &supported_systems);
+	for (int i = 0; i < UBX_M10_GNSS_SYS_CNT; ++i) {
+		gnss_systems_t system = 1 << i;
+
+		if (system & supported_systems) {
+			bool enable = (systems & system) ? UBX_CFG_GNSS_FLAG_ENABLE :
+				      UBX_CFG_GNSS_FLAG_DISABLE;
+			ubx_m10_config_block_fill(dev, system, frame_data, filled_blocks++, enable);
+		}
+	}
+
+	ret = ubx_m10_modem_ubx_script_init(dev, &script, data->ubx_frame_buf,
+		sizeof(data->ubx_frame_buf), MODEM_UBX_RETRY_DEFAULT, frame_data, UBX_CLASS_CFG,
+		UBX_CFG_GNSS, UBX_CFG_GNSS_PAYLOAD_SZ(UBX_M10_GNSS_SUPP_SYS_CNT));
+	if (ret < 0) {
+		goto out_1;
+	}
+
+	script.ubx_frame_size = ret;
+	ret = ubx_m10_modem_ubx_run_script(dev, &script);
+	if (ret < 0) {
+		goto out_1;
+	}
+
+out_1:
```

Comment:
go to should be named to reflect what it does, see [Linux coding style](https://www.kernel.org/doc/html/v4.10/process/coding-style.html#centralized-exiting-of-functions)

I would name `out_1` `free_and_unlock`, and `out_2` `unlock` This goes for the other `out`s as well I think :)